tools/sync.js

```diff
@@ -0,0 +1,214 @@
+/*
+Name: Ethereum Blockchain syncer
+Version: .0.0.1
+This file will start syncing the blockchain from the node address you provide in the conf.json file.
+Please read the README in the root directory that explains the parameters of this code
+*/
+require( '../db.js' );
+var etherUnits = require("../lib/etherUnits.js");
+var BigNumber = require('bignumber.js');
+
+var fs = require('fs');
+var Web3 = require('web3');
+
+var mongoose        = require( 'mongoose' );
+var Block           = mongoose.model( 'Block' );
+var Transaction     = mongoose.model( 'Transaction' );
+
+//Just listen for latest blocks and sync from the start of the app.
+var listenBlocks = function(config) {
```

Comment:
This section of code seems like `updateLastSynced` to me.

Right now it looks like you're trying to start from both the last synced point & grab new blocks. This is a bit fancy and probably causes duplicated effort in logging to the database at best and race conditions at worst.

I think it should be one or the other:

* Start from newest, provides fastest "ready" state but may not be too important for a block explorer since anyone launching one can afford to wait for the sync to complete
* Start from oldest, slower till "ready" but should be able to sync alongside a newly syncing node (which might be fairly common). Also more inline with how blockchains work (ie. `0 -> "latest"`) which I imagine simplifies the program logically.

Reply:
This has been completely reworked so the fanciness can stay around 💃